packages/docregistry/src/default.ts

```diff
@@ -488,23 +495,15 @@ class MimeRenderer extends Widget implements IRenderMime.IReadyWidget {
   /**
    * Render the mime content.
    */
-  private _render(): void {
+  private _render(): Promise<void> {
     let context = this._context;
     let model = context.model;
-    let layout = this.layout as PanelLayout;
-    let data: JSONObject = {};
     if (this._dataType === 'string') {
-      data[this._mimeType] = model.toString();
+      this._mimeModel.data.set(this._mimeType, model.toString());
     } else {
-      data[this._mimeType] = model.toJSON();
-    }
-    let mimeModel = new MimeModel({ data, trusted: false });
-    let widget = this._rendermime.render(mimeModel);
-    if (layout.widgets.length === 2) {
-      // The toolbar is layout.widgets[0]
-      layout.widgets[1].dispose();
+      this._mimeModel.data.set(this._mimeType, model.toJSON());
```

Comment:
Not of a fan of this mutation here, then passing the same mime model back into the renderer. If the renderer is expecting the model to be immutable, it might not re-render.